internal/controller/k8s_resources.go

```diff
@@ -99,10 +107,101 @@ func convertMetaObjectToMetadata(logger *zap.SugaredLogger, ctx context.Context,
 			return objMetadata, nil
 		}
 		objMetadata.KindSpecific = &pb.KubernetesObjectData_Node{Node: &pb.KubernetesNodeData{ProviderId: providerId}}
+	case "Service":
+		convertedServiceData, err := convertToKubernetesServiceData(ctx, obj.GetName(), clientset, obj.GetNamespace())
+		if err != nil {
+			return objMetadata, nil
+		}
+		objMetadata.KindSpecific = &pb.KubernetesObjectData_Service{Service: convertedServiceData}
+
 	}
 	return objMetadata, nil
 }
 
+func convertIngressToStringList(ingresses []v1.LoadBalancerIngress) []string {
+	result := []string{}
+	for _, ingress := range ingresses {
+		if ingress.IP != "" {
+			result = append(result, ingress.IP)
+		} else if ingress.Hostname != "" {
+			result = append(result, ingress.Hostname)
+		}
+	}
+	return result
+}
+
+func convertServicePortsToPorts(servicePorts []v1.ServicePort) []*Ports {
+	ports := make([]*Ports, 0, len(servicePorts))
+	for _, sp := range servicePorts {
+		protocol := string(sp.Protocol)
+		if protocol == "" {
+			protocol = string(v1.ProtocolTCP)
+		}
+		port := &Ports{
+			Port:     sp.Port,
+			Protocol: protocol,
+		}
+		if sp.NodePort != 0 {
+			port.NodePort = &sp.NodePort
+		}
+		ports = append(ports, port)
+	}
+	return ports
+}
+
+// Combine all IPs into a single list
+func combineIPAddresses(clusterIps []string, externalIps []string, loadBalancerIngresses []string, loadBalancerIp string) []string {
+	combinedIPs := []string{}
```

Comment:
Really minor nit. Not a PR blocker. Feel free to resolve this conversation without fixing.
```suggestion
	combinedIPs := make([]string, 0, len(clusterIps)+len(externalIps)+len(loadBalancerIngresses)+1)
```